solidity/dashboard/src/components/RecoverStakeButton.jsx

```diff
@@ -1,21 +1,28 @@
-import React, { useContext, useCallback } from "react"
+import React, { useContext, useCallback, useMemo } from "react"
 import { SubmitButton } from "./Button"
 import { Web3Context } from "./WithWeb3Context"
 import { useShowMessage, messageType } from "./Message"
-import {
-  TOKEN_GRANT_CONTRACT_NAME,
-  TOKEN_STAKING_CONTRACT_NAME,
-} from "../constants/constants"
 
 const RecoverStakeButton = ({ operatorAddress, ...props }) => {
   const web3Context = useContext(Web3Context)
-  const { yourAddress } = web3Context
+  const { yourAddress, grantContract, stakingContract } = web3Context
```

Comment:
`yourAddress` always points to the currently selected account. It can be owner/grantee/operator we don't care. The dapp always sends transactions from the current selected account. I think it is clearer than doing eg. like this `{ yourAddress: operatorAddress } = useWeb3Context()`. Also, the recovery function can be called by anyone.

Reply:
can we maybe add a comment what `yourAddress` can represent?